### DISCUSSION TITLE
Zotero internal item ID field as Citation key format variable

### DISCUSSION THREAD

**Discussion by ryanwwest** [retorquere/zotero-better-bibtex#2285] [Q&A] [answered]
I'm wondering if you can get the Zotero internal item ID field as a variable in the citation key format (https://retorque.re/zotero-better-bibtex/citing/). There is an `itemID` field available that translates to a value like `zotero-678` (I think you can get this with the variable `ArchiveID`), but this is not the item ID I'm referring to. I'd like to get the value used at the end of a select URI, e.g., `JFWCVTFR` from `zotero://select/library/items/JFWCVTFR`.

I understand that I could also just not use citekeys for my Select URLs; however, I'm nevertheless hoping to find a way to get this ID as part of my citekey. I'm open to using Postscript or other things if necessary. Thanks!

**Comment by ryanwwest** [comment]
A similar question I'm wondering which would also solve my problem is whether, when exporting citations to JSON CSL (or another file format that can stay updated with changes), I can make the `id` field of a citation equal to the Zotero internal item ID (instead of equal to `citation-key`). For example, one (simplified) line of JSON output for a citation is:
```
{
    "id":"20ThingsVe2021",
    "citation-key":"20ThingsVe2021",
    "title":"20 Things I've Learned in my 20 Years as a Software Engineer",
    "type":"webpage"
}
``` 
`id` is currently the same as `citation-key`.

**Reply by retorquere** [reply]
You can change that with a [postscript](https://retorque.re/zotero-better-bibtex/exporting/scripting/), or by having the citation key be the item key (which I can get you if you open an issue).

**Comment by retorquere** [comment, marked as answer]
This requires a change. Please open a new issue, and I'll get you something you can test.

**Reply by ryanwwest** [reply]
Awesome, thank you! New issue is [here](https://github.com/retorquere/zotero-better-bibtex/issues/2286).
